From 08e46feb6553af670754e65d94c3bb6fcd4e0cf9 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 31 Oct 2021 23:39:44 -0700
Subject: [PATCH] Fix build on mips/clang

clang13 crashes on mips, until its fixed upstream disable tailcall on
mips

https://bugs.llvm.org/show_bug.cgi?id=52367

Upstream-Status: Inappropriate [Clang workaround]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/google/protobuf/port_def.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -255,6 +255,7 @@
 #error PROTOBUF_TAILCALL was previously defined
 #endif
 #if __has_cpp_attribute(clang::musttail) && !defined(__arm__) && \
+    !defined(__mips__) && \
     !defined(_ARCH_PPC) && !defined(__wasm__) &&                 \
     !(defined(_MSC_VER) && defined(_M_IX86)) &&                  \
     !(defined(__NDK_MAJOR__) && __NDK_MAJOR <= 24)
